### ISSUE TITLE
Do-not-change option for list of fields

### ISSUE THREAD

**Issue by rpspringuel** [nschloe/betterbib#80]
Take the following entry as a sample:

```
@article{Byskov:1972,
	Author = {Byskov, Martha},
	Date-Added = {2017-10-24 00:21:40 +0000},
	Date-Modified = {2017-11-02 02:27:24 +0000},
	Journal = {Studio Theologica},
	Pages = {25--32},
	Read = {1},
	Title = {Verus Deus --- verus homo: Luc 3.23--38},
	Volume = {26},
	Year = {1972},
	Bdsk-File-1 = {YnBsaXN0MDDUAQIDBAUGJCVYJHZlcnNpb25YJG9iamVjdHNZJGFyY2hpdmVyVCR0b3ASAAGGoKgHCBMUFRYaIVUkbnVsbNMJCgsMDxJXTlMua2V5c1pOUy5vYmplY3RzViRjbGFzc6INDoACgAOiEBGABIAFgAdccmVsYXRpdmVQYXRoWWFsaWFzRGF0YV8QTS4uLy4uLy4uLy4uL0RvY3VtZW50cy9SZWZlcmVuY2VzL0J5c2tvdi9CeXNrb3YxOTcyX1ZlcnVzIERldXMgLS0tIHZlcnVzLWEucGRm0hcLGBlXTlMuZGF0YU8RAfoAAAAAAfoAAgAADE1hY2ludG9zaCBIRAAAAAAAAAAAAAAAAAAAAMzdHpxIKwAAAm6BJh9CeXNrb3YxOTcyX1ZlcnVzIERlIzI2RTgxMTEucGRmAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACboER1hP/yQAAAAAAAAAAAAQABAAACSAAAAAAAAAAAAAAAAAAAAAGQnlza292ABAACAAAzN1k7AAAABEACAAA1hQ4CQAAAAEAFAJugSYABn0oAAZsuQAGPcgAAhAqAAIAV01hY2ludG9zaCBIRDpVc2VyczoAUlBTOgBEb2N1bWVudHM6AFJlZmVyZW5jZXM6AEJ5c2tvdjoAQnlza292MTk3Ml9WZXJ1cyBEZSMyNkU4MTExLnBkZgAADgBMACUAQgB5AHMAawBvAHYAMQA5ADcAMgBfAFYAZQByAHUAcwAgAEQAZQB1AHMAIAAtAC0ALQAgAHYAZQByAHUAcwAtAGEALgBwAGQAZgAPABoADABNAGEAYwBpAG4AdABvAHMAaAAgAEgARAASAEtVc2Vycy9SUFMvRG9jdW1lbnRzL1JlZmVyZW5jZXMvQnlza292L0J5c2tvdjE5NzJfVmVydXMgRGV1cyAtLS0gdmVydXMtYS5wZGYAABMAAS8AABUAAgAK//8AAIAG0hscHR5aJGNsYXNzbmFtZVgkY2xhc3Nlc11OU011dGFibGVEYXRhox0fIFZOU0RhdGFYTlNPYmplY3TSGxwiI1xOU0RpY3Rpb25hcnmiIiBfEA9OU0tleWVkQXJjaGl2ZXLRJidUcm9vdIABAAgAEQAaACMALQAyADcAQABGAE0AVQBgAGcAagBsAG4AcQBzAHUAdwCEAI4A3gDjAOsC6QLrAvAC+wMEAxIDFgMdAyYDKwM4AzsDTQNQA1UAAAAAAAACAQAAAAAAAAAoAAAAAAAAAAAAAAAAAAADVw==}}
```
It contains 4 special fields which are related to BibDesk (the program I use to manage my bib file):

1. Date-Added: the date the entry was added to my database
2. Date-Modified: the date I last edited the entry
3. Read: this shows up as a check mark in BibDesk indicating that I have read the article
4. Bdsk-File: a file link that BibDesk has created to a pdf of the article on my computer

When I run this entry through betterbib I get the following out:

```
@article{Byskov:1972,
 author = {Byskov, Martha},
 doi = {10.1080/00393387208599926},
 issn = {0039-338X, 1502-7791},
 journal = {Studia Theologica - Nordic Journal of Theology},
 month = jan,
 number = {1},
 pages = {25-32},
 publisher = {Informa UK Limited},
 source = {Crossref},
 title = {Verus Deus ‐ veras homo Luc 3.23–381},
 url = {https://doi.org/10.1080/00393387208599926},
 volume = {26},
 year = {1972},
}
```
I'm quite happy with the added and corrected information, but the BibDesk specific fields have been lost (making the revised entry less useful, particularly because of the loss of the file link).  I would like an option which allows me to list certain fields which should be transferred from the original entry (if they occur) to the revised one without change.  In my case these would be fields which will not occur in the look-up source, but the list of fields should probably generalize to a "do-not-change" list in the case of fields which are found in the information retrieved from the source.

Further, #33 could be also be handled by a sort of inverse of this mechanism (i.e. an option in which betterbib is told which fields it is allowed to change, rather than which it is not allowed to change).

**Comment by nschloe** [comment]
I made sure now that keys which are there in the original bibtex file are preserved through the sync. This should fix it.

**Comment by rpspringuel** [comment]
While not as generally applicable as a "do-not-change" list would have been, simply preserving the keys suits my purposes.  Thanks.

**Comment by rpspringuel** [comment]
Hmm... May have spoken too soon.  It seems that some fields are "preserved" even when corrected information is found on CrossRef.  

I don't have time to work out all the details of what is going on at the moment, but my best guess is as follows:
This may have to do with case matching as field names in my original file are all initial caps, but betterbib uses all lowercase (i.e. `Author` vs. `author`).  As a result `Author` is thought to be a field which is not present in the information collected from CrossRef, but when added, it overwrites the `author` field.

If you need more information, I'll try to set aside some time this weekend to look into this further.

**Comment by rpspringuel** [comment]
Okay, what I was doing didn't take as long as I thought so here goes:

Take the following entry as the sample:

```
@book{Adamczewski_2016,
 author = {Bartosz Adamczewski},
 doi = {10.3726/978-3-653-06278-6},
 month = {jan},
 publisher = {Peter Lang},
 title = {The Gospel of Luke},
 url = {https://doi.org/10.3726%2F978-3-653-06278-6},
 year = 2016,
}
```

This entry was created by downloading it directly from the Cite function on CrossRef (so I know the information will match exactly what betterbib will find).  Run this through betterbib and it comes out:

```
@book{Adamczewski_2016,
 author = {Adamczewski, Bartosz},
 doi = {10.3726/978-3-653-06278-6},
 isbn = {9783653062786, 9783653959987, 9783653959970, 9783631667200},
 month = jan,
 publisher = {Peter Lang},
 source = {Crossref},
 title = {The Gospel of {Luke}},
 url = {http://dx.doi.org/10.3726/978-3-653-06278-6},
 year = {2016},
}
```

So far so good.  However, let's introduce an error in the entry by removing the last letter of the author's last name:

```
@book{Adamczewski_2016,
 Author = {Bartosz Adamczewsk},
 doi = {10.3726/978-3-653-06278-6},
 month = {jan},
 publisher = {Peter Lang},
 title = {The Gospel of Luke},
 url = {https://doi.org/10.3726%2F978-3-653-06278-6},
 year = 2016,
}
```
Now I get:
```
@book{Adamczewski_2016,
 author = {Adamczewsk, Bartosz},
 doi = {10.3726/978-3-653-06278-6},
 isbn = {9783653062786, 9783653959987, 9783653959970, 9783631667200},
 month = jan,
 publisher = {Peter Lang},
 source = {Crossref},
 title = {The Gospel of {Luke}},
 url = {http://dx.doi.org/10.3726/978-3-653-06278-6},
 year = {2016},
}
```
The misspelling has persisted and so betterbib hasn't actually fixed the entry properly.  Instead, all it has done is add the information which CrossRef doesn't make available in its Cite link, but which betterbib does get as part of its query.

**Comment by nschloe** [comment]
Nicely worked out. I'm looking at it now.

**Comment by nschloe** [comment]
Fixed that; update to betterbib 3.0.1.
